### PR TITLE
Travis: Upload builds to builds.citra-emu.org

### LINKED ISSUE #9: Provide binary release
There should be something to automaticly compile and provide this.

It's really not acceptable to expect people to use their legally dumped stuff if they first have to compile this yourself.
If users don't already have the 3DS toolchain installed they'd have to google for it first - instead they are much more likely to google for the dumped files (which are also probably easier to find than a binary of 3dsutils).

There is already [a folder for this on the build server](https://builds.citra-emu.org/3dsutils/nightly/) even..

---

Ideally this would be implemented using CI and automaticly push changes.
(Similar for the test cases in our other repo, but that's beyond the scope of this issue)

### PR DESCRIPTION
**Pull request by JayFoxRox**
First, this fixes the travis build by switching the directory before `make`. The current script used to compile ctrulib twice and never actually compiled 3dsutils..

This also adds the travis-build script from the Citra main repo which uploads to builds.citra-emu.org.
(This still needs proper credentials before a merge - I don't have access to travis and neither the credentials)
This is intended to fix #9 .

Lastly this improves travis build time dramaticly by caching the devkitPro installation.

---

@archshift I need to update the credentials for travis-upload, contact me on IRC? From what I gathered you are maintaining builds.citra-emu.org ? I've also contacted flamesage, but he doesn't have the necessary permissions. Also I've never worked with travis encrypted vars, so I might need some guidance.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,18 @@
 language: cpp
 
+env:
+  global:
+    - secure: "AXHFIafTmbGDsHD3mUVj5a4I397DQjti/WoqAJGUp2PglxTcc04BwxZ9Z+xLuf5N2Hs5r9ojAJLT8OGxJCLBDXzneQTNSqXbFuYSLbqrEAiIRlA9eRIotWCg+wYcO+5e8MKX+cHVKwiIWasUB21AtCdq6msh6Y3pUshZp212VPg="
```
Comment (JayFoxRox) [inline .travis.yml]: Requires change to match repo privkey

**Comment by MerryMage** [comment]
Ping @bunnei.

**Comment by JayFoxRox** [comment]
I agree with https://github.com/citra-emu/3dsutils/issues/9#issuecomment-246157143 . We don't know how long we'll continue with builds.citra-emu.org either. So going for releases is probably fine

**Comment by JayFoxRox** [comment]
Closed in favour of #13

**Pull request closed without merging**